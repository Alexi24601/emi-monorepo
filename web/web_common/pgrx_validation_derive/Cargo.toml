[package]
name = "pgrx_validation_derive"
version = "0.1.0"
edition = "2021"

[dependencies]
syn.workspace = true
quote.workspace = true
proc-macro2.workspace = true

[dev-dependencies]
trybuild.workspace = true
validation-errors.workspace = true

[lib]
proc-macro = true

[features]
pgrx = []
