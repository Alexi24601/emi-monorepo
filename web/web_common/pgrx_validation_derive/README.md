# PGRX Validation Derive

Procedural macros to dispatch the function output data type according to the features enabled.
