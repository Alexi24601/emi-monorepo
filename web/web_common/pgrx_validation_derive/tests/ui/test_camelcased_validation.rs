//! Tests whether the derive macro raises the expected error when the function name is camelcased.

use pgrx_validation_derive::validation;

#[validation]
/// A CamelCased validation function should fail.
pub fn MustNotBeEmpty(_arg: &str) -> Result<(), validation_errors::SingleFieldError> {
    Ok(())
}

/// The main for this test.
fn main() {}
