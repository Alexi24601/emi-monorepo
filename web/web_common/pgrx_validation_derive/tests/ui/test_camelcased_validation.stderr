error: Function `MustNotBeEmpty` must be snake_case to be decorated with `validation`.
 --> tests/ui/test_camelcased_validation.rs:7:8
  |
7 | pub fn MustNotBeEmpty(_arg: &str) -> Result<(), validation_errors::SingleFieldError> {
  |        ^^^^^^^^^^^^^^
